Dockerfile

```diff
@@ -17,6 +13,18 @@ RUN mv ./kubectl /usr/local/bin/kubectl && chmod +x /usr/local/bin/kubectl
 
 # Add scripts for Istio
 COPY scripts /usr/local/app/scripts/
-RUN chmod +x /usr/local/app/scripts/init_kubeconfig.sh /usr/local/app/scripts/run.sh /usr/local/app/scripts/create_istio_system.sh /usr/local/app/scripts/uninstall_istio_system.sh /usr/local/app/scripts/get_grafana_dashboards.sh
+COPY versions.txt /usr/local/app/scripts/versions.txt
+RUN chmod +x /usr/local/app/scripts/init_kubeconfig.sh /usr/local/app/scripts/run.sh /usr/local/app/scripts/create_istio_system.sh /usr/local/app/scripts/uninstall_istio_system.sh /usr/local/app/scripts/get_grafana_dashboards.sh /usr/local/app/scripts/fetch_istio_releases.sh /usr/local/app/scripts/versions.txt
 RUN mkdir -p /usr/local/app/dashboards && /usr/local/app/scripts/get_grafana_dashboards.sh
+
+# Add nginx configuration
+COPY nginx.conf /etc/nginx/nginx.conf
+
+# Get Istio tar for nginx
+RUN mkdir -p /opt/istio-releases && /usr/local/app/scripts/fetch_istio_releases.sh /opt/istio-releases
+
+RUN mkdir -p /var/cache/nginx
+RUN chmod -R 777 /var/cache/nginx /etc/ssl /var/run /usr/share/nginx /usr/local/share/ca-certificates
+
+USER nginx
```

Comment:
Where does this user get created ?

Reply:
For both of the above comments you should be creating a new user and giving them access to those directories, instead of just making it writeable for everyone.